sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/SortOrder.scala

```diff
@@ -96,6 +96,24 @@ object SortOrder {
      sameOrderExpressions: Set[Expression] = Set.empty): SortOrder = {
     new SortOrder(child, direction, direction.defaultNullOrdering, sameOrderExpressions)
   }
+
+  /**
+   * Returns if a sequence of SortOrder satisfies another sequence of SortOrder.
+   *
+   * SortOrder sequence A satisfies SortOrder sequence B if and only if B is an equivalent of A
+   * or of A's prefix.
+   */
+  def orderingSatisfies(seq1: Seq[SortOrder], seq2: Seq[SortOrder]): Boolean = {
```

Comment:
I had originally named them "actualOrdering" and "requiredOrdering" in SparkPlan, but since I've moved the function here to SortOrder, thought the names should be more general.